### PR TITLE
Introduce HandleConfigurationRequestContext.Claims and return a default list of supported protocol claims

### PR DESCRIPTION
**Pull request by kevinchalet**
Fixes https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/issues/522.

/cc @NinoFloris

### CODE REVIEW

File: src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Discovery.cs [hunk 1]
```diff
@@ -246,6 +246,13 @@ public partial class OpenIdConnectServerHandler
 
             notification.Scopes.Add(OpenIdConnectConstants.Scopes.OpenId);
 
+            notification.Claims.Add(OpenIdConnectConstants.Claims.Audience);
```
Comment (kevinchalet) [inline src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Discovery.cs:249]: @NinoFloris let me know if you can think of any other claim.

Comment (NinoFloris) [reply src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Discovery.cs:249]: ah nvm (didn't see the full diff)

Comment (kevinchalet) [reply src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Discovery.cs:249]: Already included. Expand the lines to see the complete list :sweat_smile:

Comment (NinoFloris) [reply src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Discovery.cs:249]: yep (edited comment) but in that case LGTM

**Pull request merged**